src/components/tasks/crud/TasksList.js

```diff
@@ -1,47 +1,55 @@
-import {useEffect, useState} from "react";
+import React, {useEffect, useState} from "react";
 import {Task} from "./Task";
 import {Button, Drawer, List, ListItem} from "@material-ui/core";
 import {AddCircleOutline} from "@material-ui/icons";
 import {constants} from "../../../utils/constants";
 import {createTask} from "./createTask";
-import {tasksApi} from "../../../utils/apis/tasksAPI";
+import CenteredCircularProgress from "../../commons/CenteredCircularProgress";
+import Swal from "sweetalert2";
 
-export const TasksList = ({ fetchTasks, id }) => {
+export const TasksList = ({ fetchTasks, onFetchFail, id, onCreateTask, onDeleteTask }) => {
 
     const [tasks, setTasks] = useState([])
     const [reload, setReload] = useState(false)
+    const [fetching, setFetching] = useState(true)
 
-    const addTask = () => {
-        createTask(tasks, reload, setReload)
-    }
+    const addTask = () => createTask(onCreateTask, reload, setReload)
 
     useEffect(() => {
         fetchTasks(id)
-            .then(data => setTasks(data))
-    }, [fetchTasks, id, reload])
+            .then(data => {
+                setTasks(data)
+                setFetching(false)
+            })
+            .catch(() => {
+                Swal.fire({
+                    title: "Error",
+                    text: "You cannot access this panel! Check your credentials!",
+                    icon: "error"
+                }).then(() => onFetchFail())
+            })
+    }, [fetchTasks, id, reload, setTasks, setFetching, onFetchFail])
```

Comment:
Jesteś pewny, że tyle wartości w tej tablicy ma być?

Reply:
Dla reacta się podoba jak wszystko co może tu być, to tu jest. Tak naprawdę oprócz reloada to nic się to nie zmienia, więc jest git